packages/repo/src/mst/mst.ts

```diff
@@ -544,27 +544,46 @@ export class MST {
   // @TODO write tests for these
 
   // Walk tree starting at key
-  async *walkLeavesFrom(key: string): AsyncIterable<Leaf> {
+  async *walkFrom(key: string): AsyncIterable<NodeEntry> {
+    yield this
     const index = await this.findGtOrEqualLeafIndex(key)
     const entries = await this.getEntries()
-    const prev = entries[index - 1]
-    if (prev && prev.isTree()) {
-      for await (const e of prev.walkLeavesFrom(key)) {
-        yield e
+    const found = entries[index]
+    if (found && found.isLeaf() && found.key === key) {
+      yield found
+    } else {
+      const prev = entries[index - 1]
+      if (prev) {
+        if (prev.isLeaf() && prev.key === key) {
+          yield prev
+        } else if (prev.isTree()) {
+          for await (const e of prev.walkFrom(key)) {
+            yield e
+          }
+        }
       }
     }
+
     for (let i = index; i < entries.length; i++) {
       const entry = entries[i]
       if (entry.isLeaf()) {
         yield entry
       } else {
-        for await (const e of entry.walkLeavesFrom(key)) {
-          yield e
+        for await (const node of entry.walkFrom(key)) {
+          yield node
         }
```

Comment:
I think there's a case that it's clearer as written, but I believe there is syntax specifically for this case if desired:
```ts
yield* entry.walkFrom(key)
```

Reply:
oh nice didn't know this syntax!